src/nunit.analyzers.tests/ConstraintsUsage/StringConstraintUsageAnalyzerTests.cs

```diff
@@ -0,0 +1,54 @@
+using Gu.Roslyn.Asserts;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.ConstraintUsage;
+using NUnit.Framework;
+
+namespace NUnit.Analyzers.Tests.ConstraintsUsage
+{
+    public class StringConstraintUsageAnalyzerTests
+    {
+        private static readonly DiagnosticAnalyzer analyzer = new StringConstraintUsageAnalyzer();
+
+        [TestCase(nameof(string.Contains), AnalyzerIdentifiers.StringContainsConstraintUsage, "Does.Contain")]
+        [TestCase(nameof(string.StartsWith), AnalyzerIdentifiers.StringStartsWithConstraintUsage, "Does.StartWith")]
+        [TestCase(nameof(string.EndsWith), AnalyzerIdentifiers.StringEndsWithConstraintUsage, "Does.EndWith")]
+        public void AnalyzeStringBooleanMethodPositiveAssert(string method, string analyzerId, string suggestedConstraint)
+        {
+            var testCode = TestUtility.WrapInTestMethod($@"Assert.True(↓""abc"".{method}(""ab""));");
+
+            var diagnostic = ExpectedDiagnostic.Create(analyzerId,
+                string.Format(StringConstraintUsageConstants.Message, suggestedConstraint));
+            AnalyzerAssert.Diagnostics(analyzer, diagnostic, testCode);
+        }
+
```

Comment:
Same comment as for the fixer. Do we need a test when the context is `Assert.That(...)` and not `Assert.True`?

Reply:
Wouldn't harm. Will add a bit later.